grasp_database_filename: $(find chair_manipulation_grasp_detection_advanced)/cfg/grasp_database.yaml

grasp_database_creator:
  num_sample_trials_per_model: 100
  min_num_grasps_per_model: 3
  max_num_grasps_per_model: 10
  models:
    - mesh: $(find chair_manipulation_chair_models)/models/dining_chair/meshes/dining_chair.ply
      point_cloud: $(find chair_manipulation_chair_models)/models/dining_chair/point_clouds/dining_chair.pcd

gripper:
  base_frame: robotiq_arg2f_base_link
  tcp_frame: tcp
  contact_threshold: 0.001
  # This defines the open and closed configurations for independent fingers.
  # The group names and states are defined in the .srdf file.
  finger_groups:
    - group_name: left_finger
      open_group_state: left_finger_open
      closed_group_state: left_finger_closed
    - group_name: right_finger
      open_group_state: right_finger_open
      closed_group_state: right_finger_closed

grasp_sampler:
  max_antipodal_normal_angle: 0.1
  max_antipodal_position_angle: 0.1
  max_equator_normal_angle: 0.1
  gripper_pad_distance: 0.136
  gripper_pad_length: 0.06

grasp_synthesizer:
  num_arms: 2
  friction_coefficient: 0.8
  num_fiction_edges: 8
  max_arm_radius_: 1.0
  world_frame: world
  arm_base_frames:
    - robot1_base_link
    - robot2_base_link

weights_offline:
  epsilon1: 1.0
  v1: 0.0
  distance: 1.0
  reachability: 0.0

weights_online:
  epsilon1: 0.0
  v1: 1.0
  distance: 1.0
  reachability: 1.0

point_cloud_preprocessor:
  # Leaf size of the voxel filter.
  # If this is set to 0.01 for example, it means that only one point is kept every 1x1x1cm voxel.
  voxel_leaf_size: 0.01
  # Number of neighbors to consider for the statistical outlier filter.
  mean_k: 50
  # Points with a distance larger than the standard deviation of the mean distance multiplied
  # by the following value will be considered outliers.
  stddev_mul_threshold: 3.0
  # The search radius for normal estimation.
  normal_search_radius: 0.05

mesh_reconstruction:
  # Maximum distance between connected points (maximum edge length).
  search_radius: 0.025
  # Maximum acceptable distance for a point to be considered,
  # relative to the distance of the nearest point.
  max_distance: 2.5
  # How many neighbors are searched for
  max_nearest_neighbors: 100
  # Points will not be connected to the current point
  #	if their normals deviate more than the specified angle.
  max_surface_angle: 0.78539 # 45 degrees
  # Minimum and maximum angle there can be in a triangle.
  # The first is not guaranteed, the second is.
  min_angle: 0.17453 # 10 degrees
  max_angle: 2.09439 # 120 degrees
  # If false, the direction of normals will not be taken into account
  # when computing the angle between them.
  normal_consistency: true